### PR TITLE
LX-1645 Create custom td-agent package

### PR DESCRIPTION
**Pull request by prashks**
This adds our custom td-agent packaging with fluentd and all necessary plugins needed for our Insight module.

Corresponding change to include this custom td-agent package as part of `delphix-platform` is at -
 https://github.com/delphix/delphix-platform/pull/51

Tests:
- Verified package build: `git-linux-pkg build -m pks-lx-bootstrap-new.dlpxdc.co td-agent`
- linux-pkg pre-push: http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/linux-pkg-build/job/master/job/pre-push/70/  
- linux-pkg update: http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/linux-pkg-build/job/master/job/update/10/ 
- appliance-build-orchestrator : http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/appliance-build-orchestrator-pre-push/506/console
- Created a VM based on the resulting image and the custom `td-agent` is installed by default as expected

```
psreenivasa@dlpxdc$ dc clone-latest dlpx-psreenivasa-master pks-lx-1645-new
pks-lx-1645-new - dlpx-psreenivasa-master
psreenivasa@dlpxdc$

delphix@ip-10-110-215-46:~$ uname -a
Linux ip-10-110-215-46 4.15.0-1032-aws #34-Ubuntu SMP Thu Jan 17 15:18:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
delphix@ip-10-110-215-46:~$
delphix@ip-10-110-215-46:~$
delphix@ip-10-110-215-46:~$ sudo apt show td-agent
Package: td-agent
Version: 3.3.0-delphix-2019.02.19.22       <<<<< Note, this indicates its our customized version
Status: install ok installed
Priority: extra
Section: misc
<<<snip>>>
```

### CODE REVIEW

File: packages/td-agent/config.sh [hunk 1]
```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+#
+# Copyright 2018 Delphix
```
Comment (sebroy) [inline packages/td-agent/config.sh]: 2019

File: packages/td-agent/config.sh [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	# Download dependent gems
+	#
+	logmust cd "$WORKDIR/repo"
+	# Ensure all required gems are installed
+	logmust bundle install --binstubs
+	# Download dependent gems using downloader
+	logmust bin/gem_downloader core_gems.rb
+	logmust bin/gem_downloader delphix_plugin_gems.rb
+	# Create required directory and add permission
+	logmust sudo mkdir -p /opt/td-agent /var/cache/omnibus
```
Comment (sebroy) [inline packages/td-agent/config.sh]: These are required by the build?

Comment (prashks) [reply packages/td-agent/config.sh]: Yes these are the required steps to build this package.

**Comment by prashks** [comment]
cc @pzakha

File: packages/td-agent/config.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# shellcheck disable=SC2034
+
+#
```
Comment (pzakha) [inline packages/td-agent/config.sh]: Here and elsewhere, please remove all the comments that came from the config.sh template.

Comment (prashks) [reply packages/td-agent/config.sh]: Thanks, removed them.

**Pull request merged**